packages/utils/test/asAbortable.test.ts

```diff
@@ -0,0 +1,58 @@
+import { AbortError, asAbortable } from '../src/asAbortable'
+
+const sleep = (ms: number) => new Promise<string>((resolve) => setTimeout(() => resolve('foobar'), ms))
+
+const TIME_UNIT = 10
+
+describe(asAbortable, () => {
+    it('works without abortController', async () => {
+        const actual = await asAbortable(
+            sleep(TIME_UNIT)
+        )
+        expect(actual).toEqual('foobar')
+    })
+
+    it('resolves if no abort controller signalled', async () => {
+        const actual = await asAbortable(
+            sleep(TIME_UNIT),
+            new AbortController()
+        )
+        expect(actual).toEqual('foobar')
+    })
+
+    it('rejects if abort controller signalled before given promise resolves', async () => {
+        const abortController = new AbortController()
+        setTimeout(() => {
+            abortController.abort()
+        }, TIME_UNIT)
+        const actual = asAbortable(
+            sleep(2 * TIME_UNIT),
+            abortController,
+            'customError'
+        )
+        return expect(actual).rejects.toEqual(new AbortError('customError'))
```

Comment:
Note that these kind of assertions don't actually test that we throw `AbortError` (see https://github.com/streamr-dev/network-monorepo/pull/925). These  just check that we throw a correct error message.

Reply:
Separate PR